docs/README.md

```diff
@@ -0,0 +1,272 @@
+# memeid
+
+`memeid` is a Scala library for generating [RFC-compliant](https://www.ietf.org/rfc/rfc4122.txt) Universal Unique Identifiers (UUIDs).
+
+> A universally unique identifier (UUID) is a 128-bit number used to identify information in computer systems.
+>
+> When generated according to the standard methods, UUIDs are for practical purposes unique. Their uniqueness does not depend on a central registration authority or coordination between the parties generating them, unlike most other numbering schemes.
+-- [Wikipedia article on UUIDs](https://en.wikipedia.org/wiki/Universally_unique_identifier)
+
+## Rationale
+
+The UUID type that ships with the JVM `java.util.UUID` has a number of problems, namely:
+
+ - A bug in the comparison function that will never be fixed https://bugs.java.com/bugdatabase/view_bug.do?bug_id=7025832
+ - Only provides UUID generation for random (V4) and non-namespaced pseudo-V3 UUIDs
+
+This library aims to solve the aforementioned issues and provide an RFC-compliant `UUID` type with coherent comparison, a rich API to get its different fields and constructors for the UUID variants.
+
+## Install
+
+Add this to your `build.sbt` file:
+
+```scala
+libraryDependencies += "com.47deg" % "memeid" % "@VERSION@"
+```
+
+## UUID construction
+
+### Time-based (v1)
+
+The time-based (V1) variant of UUIDs is the fastest to generate. It uses a monotonic clock and node information to generate UUIDs.
+
+```scala mdoc
+import memeid.UUID
+
+UUID.V1.next
+UUID.V1.next
+UUID.V1.next
+```
+
+### Random (v4)
+
+The cryptographically random variant, equivalent to `java.util.UUID/randomUUID`.
+
+```scala mdoc
+import memeid.UUID
+
+UUID.V4.random
+UUID.V4.random
+UUID.V4.random
+```
+
+### Namespaced (v3, v5)
+
+Namespaced UUIDs are generated from a UUID (namespace) and a hashed value (name), V3 uses MD5 and V5 uses SHA1 hash.
+
+```scala mdoc
+import memeid.UUID
+
+val namespace = UUID.V1.next
+
+UUID.V3(namespace, "my-secret-code")
+UUID.V5(namespace, "my-secret-code")
+```
+
+If you want to hash a custom type, you must provide an implicit `memeid.digest.Digestible` instance. This instance is used to convert your type into a byte array for hashing.
+
+```scala mdoc
+import memeid.digest.Digestible
```

Comment:
I would split the pre-conditions and the UUID creation in two blocks, being the first one an [`mdoc:silent` doc](https://scalameta.org/mdoc/docs/modifiers.html#silent)